src/uikit.cs

```diff
@@ -21156,4 +21200,167 @@ interface UIKey : NSCopying, NSCoding {
+
+		[Export ("invalidate")]
+		void Invalidate ();
+	}
+
+	interface IUIPointerInteractionDelegate { }
+
+	[NoWatch, NoTV, iOS (13,4)]
+	[Protocol, Model (AutoGeneratedName = true)]
+	[BaseType (typeof (NSObject))]
+	interface UIPointerInteractionDelegate {
+
+		[Export ("pointerInteraction:regionForRequest:defaultRegion:")]
+		[return: NullAllowed]
+		UIPointerRegion GetRegionForRequest (UIPointerInteraction interaction, UIPointerRegionRequest request, UIPointerRegion defaultRegion);
+
+		[Export ("pointerInteraction:styleForRegion:")]
+		[return: NullAllowed]
+		UIPointerStyle GetStyleForRegion (UIPointerInteraction interaction, UIPointerRegion region);
+
+		[Export ("pointerInteraction:willEnterRegion:animator:")]
+		void WillEnterRegion (UIPointerInteraction interaction, UIPointerRegion region, IUIPointerInteractionAnimating animator);
+
+		[Export ("pointerInteraction:willExitRegion:animator:")]
+		void WillExitRegion (UIPointerInteraction interaction, UIPointerRegion region, IUIPointerInteractionAnimating animator);
+	}
+
+	[NoWatch, NoTV, iOS (13,4)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface UIPointerRegionRequest {
+		
+		[Export ("location")]
+		CGPoint Location { get; }
+
+		[Export ("modifiers")]
+		UIKeyModifierFlags Modifiers { get; }
+	}
+
+	interface IUIPointerInteractionAnimating { }
+	
+	[NoWatch, NoTV, iOS (13,4)]
+	[Protocol]
+	interface UIPointerInteractionAnimating {
+
+		[Abstract]
+		[Export ("addAnimations:")]
+		void AddAnimations (Action animations);
+
+		[Abstract]
+		[Export ("addCompletion:")]
+		void AddCompletion (Action<bool> completion);
+	}
+
+	[NoWatch, NoTV, iOS (13,4)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface UIPointerRegion : NSCopying {
+
+		[Export ("rect")]
+		CGRect Rect { get; }
+
+		[NullAllowed, Export ("identifier")]
+		NSObject Identifier { get; }
+
+		[Static]
+		[Export ("regionWithRect:identifier:")]
+		UIPointerRegion Create (CGRect rect, [NullAllowed] NSObject identifier);
+	}
+
+	[NoWatch, NoTV, iOS (13,4)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface UIPointerStyle : NSCopying {
+
+		[Static]
+		[Export ("styleWithEffect:shape:")]
+		UIPointerStyle Create (UIPointerEffect effect, [NullAllowed] UIPointerShape shape);
+
+		[Static]
+		[Export ("styleWithShape:constrainedAxes:")]
+		UIPointerStyle Create (UIPointerShape shape, UIAxis axes);
+
+		[Static]
+		[Export ("hiddenPointerStyle")]
+		UIPointerStyle CreateHiddenPointerStyle ();
+	}
+
+	[NoWatch, NoTV, iOS (13,4)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface UIPointerEffect : NSCopying {
+
+		[Export ("preview", ArgumentSemantic.Copy)]
+		UITargetedPreview Preview { get; }
+
+		[Static]
+		[Export ("effectWithPreview:")]
+		UIPointerEffect Create (UITargetedPreview preview);
+	}
+
+	[NoWatch, NoTV, iOS (13,4)]
+	[BaseType (typeof (UIPointerEffect))]
+	interface UIPointerHighlightEffect {
+
+	}
```

Comment:
any reason not to have `[DisableDefaultCtor]` ?

Reply:
I know that parent `UIPointerEffect` has `DisableDefaultCtor` but then this is unusable same for `UIPointerLiftEffect` and `UIPointerHoverEffect` and specially the last one has getters and setters so looks like init is valid here

Reply:
ok, the usage is clearer with the documentation
```
* @discussion UIPointerEffect attempts to determine the appropriate effect for the given preview automatically.
*             Use one of its subclasses to request a specific system-provided effect.
```